corehq/apps/hqcase/api/get_list.py

```diff
@@ -72,15 +76,22 @@ def _to_date_filters(field, date_filter):
 
 
 def get_list(domain, params):
+    if 'cursor' in params:
+        params_string = b64decode(params['cursor']).decode('utf-8')
+        params = QueryDict(params_string).dict()
+
     es_result = _run_query(domain, params)
     hits = es_result.hits
     ret = {
         "matching_records": es_result.total,
         "cases": [serialize_es_case(case) for case in hits],
     }
+
     cases_in_result = len(hits)
     if cases_in_result and es_result.total > cases_in_result:
-        ret['next'] = {**params, **{'indexed_on.gte': hits[-1]["@indexed_on"]}}
+        cursor = urlencode({**params, **{'indexed_on.gte': hits[-1]["@indexed_on"]}})
```

Comment:
just confirming that this will include any filters that I applied in the first place right? So my requests might look like this:

1st request:
    http://localhost:8000/a/esoergel/api/v0.6/case/?type=ABC&owner_id=123

Next url:

    http://localhost:8000/a/esoergel/api/v0.6/case/?cursor=XXXX  

Where XXXX includes the `indexed_on.gte` param as well as the `type` and `owner_id` params from the first URL


```

Reply:
That's correct, it encodes everything in there.  I was unsure whether to do that or to leave the other parameters visible and only encode the pagination param `indexed_on.gte`.  This way seemed simpler from the user's perspective and gives us a bit more freedom to change things later if we needed.  I'm still not too wedded to the approach and would be open to other considerations.